SingularityService/src/main/java/com/hubspot/singularity/config/SingularityConfiguration.java

```diff
@@ -261,6 +261,8 @@
 
   private long persistHistoryEverySeconds = TimeUnit.MINUTES.toSeconds(2);
 
+  private int historyPollerConcurrency = 5;
```

Comment:
if each history persister is using the same executor, could one end up backing up the others if we only allow 5 concurrent items to be saved at a time?

Reply:
No, each runs behind the persister lock already